alchemy-web/src/content/docs/providers/cloudflare/worker.mdx

```diff
@@ -5,80 +5,179 @@ description: Learn how to deploy, configure, and manage Cloudflare Workers using
+  compatibilityFlags: ["nodejs_compat"],  // Low-level runtime flags; consider using `compatibility` preset instead
+  compatibility: "node",                  // Compatibility preset (optional) — expands flags for common use-cases like Node.js compatibility
+  adopt: true,                            // Adopt existing Worker if present; falls back to scope.adopt when undefined
   bindings: {
-    CACHE: kv,
-    USERS: users,
+    CACHE: cache,                         // KV Namespace binding
+    STORAGE: storage,                     // R2 bucket binding
+    API_KEY: alchemy.secret("secret-value") // Plain-text env binding (converted to plain_text)
+    // Other binding types include: DurableObjectNamespace, Queue, Secret, Assets, Container, Workflow
+  },
+  observability: {
+    enabled: true                         // Enable worker logs / observability (default: true)
+  },
+  // Upload source maps to improve stack traces (set to false to disable)
+  sourceMap: true,
+  // Optional runtime limits — increase only when necessary after monitoring
+  limits: {
+    cpu_ms: 50_000                        // Max CPU time in ms (default: 30_000)
   },
+  // Other Advanced 🧪 & commonly used options
+  placement: { mode: "smart" },          // Smart placement to optimize network placement for latency
+  assets: { path: "./public", run_worker_first: false }, // Static assets configuration
+  crons: ["0 0 * * *"],                  // Scheduled cron triggers (standard cron syntax)
+  eventSources: [                        // Background event sources (queues, streams)
+    { queue: taskQueue, settings: { batchSize: 15, maxConcurrency: 3 } }
+  ],
+  routes: ["api.example.com/*"],          // Route patterns to bind the worker
+  domains: ["example.com"],               // Custom domains to bind (can be objects with zoneId/adopt)
+  version: "pr-123",                      // Publish as preview version with label
+  dev: { port: 8787 },                    // Local development options (port, remote, tunnel)
+  tunnel: true,                           // Expose the worker via a Cloudflare Tunnel.
+  namespace: await DispatchNamespace("my-dispatch-ns", {
+    namespace: "production-dispatch"
+  }),                    // Deploy to dispatch namespace (string | DispatchNamespace)
+  noBundle: false,                        // Disable bundling (deploy files as-is)
+  rules: [{ globs: ["**/*.wasm"] }],      // Additional bundle rules
 });
 ```
 
-Access in `./src/api.ts` by importing `cloudflare:workers`:
+:::tip
+Monitor your CPU usage with Cloudflare Analytics to understand if increased limits are necessary. Start with default limits and scale up only when needed to optimize costs.
+:::
 
+Access bindings using modern `env` import or traditional parameter approach:
 ```ts
 import { env } from "cloudflare:workers";
 
 export default {
-  async fetch() {
+  async fetch(request: Request, env: Env) {
```

Comment:
Where is the `Env` type defined?

Reply:
Should we just introduce `typeof myWorker.Env` immediately? This extra step leaves out important information and isn't the pattern we recommend.

Reply:
This was suppose to be showing the import `env` from Workers, but if we want to just forgo showing this at all and consider it an Alchemy anti-pattern (we lose type inference) I am 100% on board with that!